src/Microsoft.Extensions.CodeGeneration.Utils/DotNet/DefaultAssemblyLoadContext.cs

```diff
@@ -144,38 +183,74 @@ public Assembly LoadFromName(AssemblyName AssemblyName)
             }
             return Load(AssemblyName);
         }
+    }
+#else
+    public partial class DefaultAssemblyLoadContext : ICodeGenAssemblyLoadContext
+    {
+        static DefaultAssemblyLoadContext()
+        {
+            NativeLibraryExtensions = new[] { ".dll" };
+        }
 
-        public static ICodeGenAssemblyLoadContext CreateAssemblyLoadContext(string nugetPackageDir)
+        public Assembly LoadFromPath(AssemblyName assemblyName, string path)
         {
-            List<string> searchPaths = new List<string>();
-            if (Directory.Exists(nugetPackageDir))
+            if (path == null)
             {
-                Queue<string> queue = new Queue<string>();
-                queue.Enqueue(nugetPackageDir);
-                while (queue.Count > 0)
+                throw new ArgumentNullException(nameof(path));
+            }
+            if(assemblyName == null)
+            {
+                throw new ArgumentNullException(nameof(assemblyName));
+            }
+            foreach(var extension in ManagedAssemblyExtensions)
+            {
+                var resolvedPath = Path.Combine(path, assemblyName.Name + extension);
+                if(File.Exists(resolvedPath))
                 {
-                    var current = queue.Dequeue();
-                    searchPaths.Add(current);
-                    try
-                    {
-                        var subdirs = Directory.EnumerateDirectories(current);
-                        if (subdirs != null)
-                        {
-                            foreach (var sd in subdirs)
-                            {
-                                queue.Enqueue(sd);
-                            }
-                        }
-                    }
-                    catch
-                    {
-                        // Do not want to fail if we cannot access certain directories.
-                        continue;
-                    }
+                    return Assembly.LoadFile(resolvedPath);
                 }
             }
+            throw new FileNotFoundException(string.Format("Could not find assembly {0} in path {1}",assemblyName.Name, path));
+        }
 
-            return new DefaultAssemblyLoadContext(null, null, searchPaths);
+        public Assembly LoadFromPath(string path)
```

Comment:
Yup, I will remove all the checks around `Assembly.LoadFile()` or just use `Assembly.LoadFrom()` here